packages/frontend2/src/app/_components/tabs.tsx

```diff
@@ -1,10 +1,61 @@
 'use client'
 
 import * as TabsPrimitive from '@radix-ui/react-tabs'
+import { usePathname, useRouter, useSearchParams } from 'next/navigation'
 import * as React from 'react'
 import { cn } from '~/utils/cn'
 
-const Tabs = TabsPrimitive.Root
+/**
+ * This component is a wrapper around the Radix Tabs component that allows you
+ * to store the selected tab in the URL search params.
+ */
+const Tabs = React.forwardRef<
+  React.ElementRef<typeof TabsPrimitive.Root>,
+  React.ComponentPropsWithoutRef<typeof TabsPrimitive.Root> & {
+    storeInSearchParams?: boolean
+  }
+>(
+  (
+    { defaultValue: passedDefaultValue, storeInSearchParams = true, ...props },
+    ref,
+  ) => {
+    const router = useRouter()
+    const pathname = usePathname()
+    const searchParams = useSearchParams()
+
+    const defaultValue = searchParams.get('tab') ?? passedDefaultValue
+
+    const onValueChange = storeInSearchParams
+      ? (value: string) => {
+          const search = new URLSearchParams(searchParams)
+
+          if (value !== passedDefaultValue) {
+            search.set('tab', value)
+          } else {
+            search.delete('tab')
+          }
+
+          const stringified = search.toString()
+
+          if (stringified) {
+            router.replace(`${pathname}?${stringified}`)
+          } else {
+            router.replace(pathname)
+          }
+        }
+      : undefined
```

Comment:
I would say that we will also preserve filters etc in query params so lets create `useSearchParamsState` or smth that will look like that:
`const [param, setParam] = useSearchParamsState('tab', passedDefaultValue)`

Reply:
done